src/contribute/precommit-triage-guide.md

```diff
@@ -0,0 +1,137 @@
+---
+layout: section
+title: "Pre-commit Slowness Triage Guide"
+permalink: /contribute/precommit-triage-guide/
+section_menu: section-menu/contribute.html
+---
+<!--
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Pre-commit Slowness Triage Guide
+
+Beam pre-commit jobs are suites of tests run automatically on Jenkins build
+machines for each pull request (PR) submitted to
+[apache/beam](https://github.com/apache/beam). For more information and the
+difference between pre-commits and post-commits, see
+[testing](/contribute/testing/).
+
+## What are fast pre-commits?
+
+Pre-commit tests are required to pass before a pull request (PR) is merged.
+When these tests are slow they slow down Beam's development process.
+The aim is to have 95% of pre-commit jobs complete within 30 minutes
+(failing or passing).
+
+Technically, the 95th percentile of running time should be below 30 minutes over
+the past 4 weeks, where running time is the duration of time the job spends in
+the Jenkins queue + the actual time it spends running.
+
+## Determining Slowness
+
+There are two main signs of slowness:
+
+1. Pre-commit jobs are timing out after 30 minutes. This can be determined from
+   the console log of a job.
+1. Pre-commits aren't timing out, but the total wait time for pre-commit results
+   is >30m. This can be determined using a Jupyter notebook.
+
+### Notebook
+
+The Beam repo contains a [Jupyter
+notebook](https://github.com/apache/beam/tree/master/.test-infra/jupyter) named
+`precommit_job_times.ipynb` that can be used to gather and visualize statistics
+about pre-commit running times.
+
+Run the notebook. It should output a table with running times. The numbers in
+the column `totalDurationMinutes_all` and the rows with a `job_name` like `4
+weeks 95th` contain the target numbers for determining slowness.
+If any of these numbers are above 30, triaging is required.
+
+#### Example
+Here's an example table of running times:
+
+![example pre-commit duration table](/images/precommit_durations.png)
+
+(Note that this example was created when pre-commits did not have 30m timeouts.)
+
+In this example, Go pre-commits are taking approximately 14 minutes, which is
+fast. Java and Python pre-commits are taking 78 and 32 minutes respectively,
+which is slow. Both Java and Python pre-commits require triage.
+
+## Triage Process
+
+1. [Search for existing
+   issues](https://issues.apache.org/jira/issues/?filter=12344461)
```

Comment:
This filter is private.
You can put full query link here and it will work for everyone. Example:
https://issues.apache.org/jira/issues/?jql=project%20%3D%20BEAM%20AND%20status%20in%20(Open%2C%20%22In%20Progress%22%2C%20Reopened)%20AND%20resolution%20%3D%20Unresolved%20AND%20component%20%3D%20test-failures%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC

Reply:
Done, changed to long-form link.
Apparently I don't have the right permissions to globally share a filter. /shrug